Steps4Impact/BadgesV2/BadgeCell.swift

```diff
@@ -0,0 +1,158 @@
+//
+//  BadgeCell.swift
+//  Steps4Impact
+//
+//  Created by Aalim Mulji on 11/9/19.
+//  Copyright © 2019 AKDN. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class BadgeCell: UICollectionViewCell {
+
+  let dateFormatter : DateFormatter = {
+    let formatter = DateFormatter()
+    formatter.dateStyle = .long
+    formatter.timeStyle = .none
+    return formatter
+  }()
+
+  var badge : Badge? {
+    didSet {
+
+      if let badge = badge {
+
+        var isDateAvailable : Bool = false
+        var formattedDateString : String = ""
+        if let date = badge.date {
+          isDateAvailable = true
+          formattedDateString = dateFormatter.string(from: date)
+        }
+        badgeImageView.isHidden = false
+        badgeLabel.isHidden = false
+        finalMedalImageView.isHidden = true
+        finalMedalLabel.isHidden = true
+        switch badge.badgeType {
+        case .steps:
+          badgeLabel.text = #"\#(badge.stepsCompleted) steps completed \#(isDateAvailable ? "on \(formattedDateString)" : "")"#
+          break
+        case .streak:
+          badgeLabel.text = #"\#(badge.streak) days streak completed \#(isDateAvailable ? "on \(formattedDateString)" : "")"#
+          break
+        case .personalProgress:
+          badgeLabel.text = #"Completed \#(badge.personalProgress) miles \#(isDateAvailable ? "on \(formattedDateString)" : "")"#
+          break
+        case .teamProgress:
+          badgeLabel.text = "Team completed \(badge.teamProgress)% of journey"
+          break
+        case .finalMedal:
+          badgeImageView.isHidden = true
+          badgeLabel.isHidden = true
+          finalMedalImageView.isHidden = false
+          finalMedalLabel.isHidden = false
+          finalMedalLabel.text = "\(badge.finalMedalAchieved.rawValue) Level Badge"
+          break
+        default:
+          badgeLabel.text = ""
+          break
+        }
+
+      }
+    }
+  }
+
+  override init(frame: CGRect) {
+    super.init(frame: frame)
+    setupViews()
+  }
+
+  let badgeImageView: UIImageView = {
+    let imageView = UIImageView()
+    imageView.image = UIImage(named: "badge_icon")
+    imageView.contentMode = .scaleAspectFit
+    imageView.translatesAutoresizingMaskIntoConstraints = false
+    imageView.layer.borderWidth = 1
+    imageView.layer.borderColor = UIColor(hex: 0x00A85D).cgColor
+    imageView.layer.cornerRadius = 36
+    return imageView
+  }()
+
+  let finalMedalImageView: UIImageView = {
+     let imageView = UIImageView()
+     imageView.image = UIImage(named: "badge_icon")
+     imageView.contentMode = .scaleAspectFit
+     imageView.translatesAutoresizingMaskIntoConstraints = false
+     imageView.layer.borderWidth = 1
+    imageView.layer.borderColor = UIColor(hex: 0x00A85D).cgColor
+     imageView.layer.cornerRadius = 75
+     return imageView
+   }()
+
+    let finalMedalLabel: UILabel = {
```

Comment:
use UILabel(typography: ) helpers so text styles can be consistent. Most of this config is done in them.